source/shared/core_results.cpp

```diff
@@ -160,6 +159,27 @@ SQLRETURN copy_buffer( _Out_ void* buffer, SQLLEN buffer_length, _Out_ SQLLEN* o
 }
 #endif // !_WIN32
 
+
+#ifndef _WIN32
+
+size_t charFromUtf16(const WCHAR src, char * dest, size_t cchDest, DWORD * pErrorCode)
+{
+    return SystemLocale::FromUtf16( CP_ACP, &src, 1, dest, cchDest, NULL, pErrorCode );
```

Comment:
it's the destination CP and internally it is changed to UTF8.
charFromUtf16 is used to get around string_to_number compile errors.